src/main/java/com/epam/esm/web/dto/GiftCertificateDto.java

```diff
@@ -0,0 +1,123 @@
+package com.epam.esm.web.dto;
+
+import com.epam.esm.model.GiftCertificate;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.RepresentationModel;
+
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class GiftCertificateDto extends RepresentationModel<GiftCertificateDto> {
+
+    private int id;
+    private String name;
+    private String description;
+    private double price;
+    private LocalDateTime createDate;
+    private LocalDateTime lastUpdateDate;
+    private int duration;
+    private Set<EntityModel<TagDto>> tags = new HashSet<>();
+
+    public static GiftCertificateDto of(GiftCertificate giftCertificate) {
```

Comment:
Use converter for such purposes. Why such functionality is implemented in Data Transfer Object